hail/src/main/scala/is/hail/rvd/AbstractRVDSpec.scala

```diff
@@ -75,55 +76,77 @@ object RVDSpec {
         }
       }
 
-    val spec = UnpartitionedRVDSpec(rowType, codecSpec, Array("part-0"))
+    val spec = OrderedRVDSpec(
+      rowType.physicalType,
+      FastIndexedSeq(),
+      codecSpec,
+      Array("part-0"),
+      RVDPartitioner.unkeyed(1))
     spec.write(hConf, path)
 
     Array(part0Count)
   }
 }
 
-abstract class RVDSpec {
-  def read(hc: HailContext, path: String, requestedType: TStruct): RVD
-
-  def readLocal(hc: HailContext, path: String, requestedType: TStruct): IndexedSeq[Row]
-
-  def write(hadoopConf: hadoop.conf.Configuration, path: String) {
-    hadoopConf.writeTextFile(path + "/metadata.json.gz") { out =>
-      implicit val formats = RVDSpec.formats
-      Serialization.write(this, out)
-    }
+object OrderedRVDSpec {
+  def apply(rowType: PStruct,
+    key: IndexedSeq[String],
+    codecSpec: CodecSpec,
+    partFiles: Array[String],
+    partitioner: RVDPartitioner): AbstractRVDSpec = {
+
+    OrderedRVDSpec(
+      RVDType(rowType, key),
+      codecSpec,
+      partFiles,
+      JSONAnnotationImpex.exportAnnotation(
+        partitioner.rangeBounds.toFastSeq,
+        partitioner.rangeBoundsType))
   }
 }
 
-case class UnpartitionedRVDSpec(
-  rowType: TStruct,
-  codecSpec: CodecSpec,
-  partFiles: Array[String]) extends RVDSpec {
-  def read(hc: HailContext, path: String, requestedType: TStruct): RVD =
-    RVD.unkeyed(
-      requestedType.physicalType,
-      hc.readRows(path, rowType, codecSpec, partFiles, requestedType))
-
-  def readLocal(hc: HailContext, path: String, requestedType: TStruct): IndexedSeq[Row] =
-    RVDSpec.readLocal(hc, path, rowType, codecSpec, partFiles, requestedType)
-}
-
 case class OrderedRVDSpec(
   rvdType: RVDType,
```

Comment:
Just a comment, the future version of this shouldn't have an RVDType, which has a physical type, but just keys and the encoded type (separate ... or ERVDType).

Reply:
yes, but I decided not to bump the file version to get closer to that.